lib/web/resources.go

```diff
@@ -351,7 +407,99 @@ func listResources(clt resourcesAPIGetter, r *http.Request, resourceKind string)
 		UseSearchAsRoles:    values.Get("searchAsRoles") == "yes",
 	}
 
-	return clt.ListResources(r.Context(), req)
+	var response *types.ListResourcesResponse
+	nextKey, left := startKey, limit
+
+	for {
+		moreResponse, err := fetchMoreResources(r.Context(), clt, &req, nextKey, left)
+		if err != nil || moreResponse.Response == nil {
+			return response, err
+		}
+
+		nextKey = moreResponse.Response.NextKey
+
+		if response != nil {
+			response.Resources = append(response.Resources, moreResponse.Response.Resources...)
+			response.NextKey = nextKey
+		} else {
+			response = moreResponse.Response
+		}
+
+		resourceLen := int32(len(response.Resources))
+		left = limit - resourceLen
+
+		if resourceLen >= limit || !moreResponse.HasMore || nextKey == "" {
+			break
+		}
+	}
+
+	return response, err
+}
+
+type fetchMoreResponse struct {
+	Response *types.ListResourcesResponse
+	HasMore  bool
+}
+
+func fetchMoreResources(
+	ctx context.Context,
+	clt resourcesAPIGetter,
+	previousReq *proto.ListResourcesRequest,
+	nextKey string,
+	left int32,
+) (fetchMoreResponse, error) {
+	req := proto.ListResourcesRequest{
+		ResourceType:        previousReq.ResourceType,
+		Limit:               left,
+		StartKey:            nextKey,
+		SortBy:              previousReq.SortBy,
+		PredicateExpression: previousReq.PredicateExpression,
+		SearchKeywords:      previousReq.SearchKeywords,
+		UseSearchAsRoles:    previousReq.UseSearchAsRoles,
+	}
+
+	response, err := clt.ListResources(ctx, req)
+	if err != nil || response == nil {
+		return fetchMoreResponse{Response: response}, err
+	}
+
+	hasMore := len(response.Resources) > 0
+	response = removeTeleportInternalResources(response)
+
+	return fetchMoreResponse{Response: response, HasMore: hasMore}, err
+}
+
+func removeTeleportInternalResources(response *types.ListResourcesResponse) *types.ListResourcesResponse {
```

Comment:
I think you're still misunderstanding the requirements here. 

The issue to to hide the _labels_ themselves, not hide resources that have internal labels.